### DISCUSSION TITLE
Get Solution to generate necessary .dll files upon compilation

### DISCUSSION THREAD

**Discussion by Codinablack** [premake/premake-core#2197] [Q&A]
I have searched everywhere in the documentation and I can't seem to find where this would be a setting or command or method. I have started using Premake to get rid of CMake obviously, and that means I am working with the source of a project that has already been being used for awhile, and was using CMake. The solution that the CMake generates, when compiled with visual studio, generates many .dll files, all of which are necessary to run the executable. 

While using Premake and Conan instead of CMake and VCPKG, the solution only generates 1 single .dll file (lua.dll) and one .exe file, but does not create the other many .dll files I am supposed to get.

Can someone explain to me or give me an example, or point me to documentation or tutorial on how to make sure the generated solution when compiled generates all the necessary .dll files as well. Please and thank you!

PS. I would also like to know how to generate the other missing solution files like, debug.props, release.props, settings.props, ect.

**Comment by redorav** [comment]
Are you able to post your premake.lua? Or at least the relevant parts that don't work for you

**Comment by Codinablack** [comment]
Well github says it doesn't support the file type, so I will just embed it here, as its not that long.

```lua
include "packages/conandeps.premake5.lua"

workspace "Titan"
   configurations { "Debug", "Release"}
   platforms { "64", "32"}
   location "./projectfiles"
   editorintegration "On"

   project        "Titan"
      kind        "ConsoleApp"
      language    "C++"
      cppdialect  "C++20"
      targetdir   "build/%{cfg.buildcfg}/bin"
      objdir      "build/%{cfg.buildcfg}/obj"
      location    "./projectfiles"
      files { "src/**.cpp", "src/**.h" }
      flags {"LinkTimeOptimization", "MultiProcessorCompile"}
      vectorextensions {"AVX"}

      filter "configurations:Debug"
         defines { "DEBUG" }
         symbols "On"
         optimize "Debug"
      filter {}

      filter "configurations:Release"
         defines { "NDEBUG" }
         symbols "On"
         optimize "Speed"
      filter {}

      filter "platforms:32"
         architecture "x86"
      filter {}

      filter "platforms:64"
         architecture "amd64"
      filter {}

      filter "system:not windows"
         buildoptions { "-Wall", "-Wextra", "-Wnon-virtual-dtor", "-Wold-style-cast", "-pedantic", "-Werror", "-pipe", "-fvisibility=hidden" }
      filter {}

      filter "system:windows"
         openmp "On"
         characterset "MBCS"
      filter {}

      filter "toolset:gcc"
         buildoptions { "-fno-strict-aliasing" }
      filter {}

      filter "toolset:clang"
         buildoptions { "-Wimplicit-fallthrough", "-Wmove" }
      filter {}

      filter { "system:macosx", "action:gmake" }
         buildoptions { "-fvisibility=hidden" }   
      filter {}

      conan_setup()
      linkoptions {"/IGNORE:4099"}
      --characterset "MBCS"
      intrinsics   "On"


```

**Comment by Codinablack** [comment]
> Are you able to post your premake.lua? Or at least the relevant parts that don't work for you
I forgot to hit reply. I shared the script I am using currently above.

**Comment by Codinablack** [comment]
Its weird, I changed symbols to "Full" and now I get an extra .dll for zlib, so I get two .dll files total now, lua's and zlibs, but in fact I need many more for mysql, boost (many components), cryptopp, pugixml, and others.
